src/components/App/app.js

```diff
@@ -7,26 +7,87 @@ import Footer from "../Footer";
 
 const App = () => {
 
+   let minId = 1;
+
+
+   const createTodo = (label) => {
+      return {
+         label: label,
+         isCompleted: false,
+         isEditing: false,
+         id: minId++
+      }
+   };
+
    const [todos, setTodos] = useState([
-      {label: 'Completed task', id: 1},
-      {label: 'Editing task', id: 2},
-      {label: 'Active task', id: 3},
+      createTodo('Completed task'),
+      createTodo('Editing task'),
+      createTodo('Active task')
    ]);
 
+   console.log(todos);
+
+   const [filter, setFilter] = useState('all');
+   
+
+   const filterTodos = todos.filter((el) => {
+      if(filter === "active") return !el.isCompleted;
+      if(filter === "completed") return el.isCompleted;
+      return true
+   });
+
+
+   const addTask = (label) => {
+      setTodos((prevState) => [...prevState, createTodo(label)])
+   };
+   
+
+   const onCompletedClick = (id) => {
+      setTodos((prevState) => 
+         prevState.map((el) => el.id === id ? {...el, isCompleted: !el.isCompleted} : el)
+      );
+    };
+
+    const onEditingClick = (id) => {
+      setTodos((prevState) => 
+         prevState.map((el) => el.id === id ? {...el, isEditing: !el.isEditing} : {...el, isEditing: false}))
+    };
+
    const deleteTask = (id) => {
-      const newTodos = todos.filter((el) => el.id !== id);
-      setTodos(newTodos);
+      setTodos((prevState) => prevState.filter((el) => el.id !== id));
+   };
+
+
+   const clearCompleted = () => {
+      setTodos((prevState) => prevState.filter((el) => !el.isCompleted));
    };
 
+   const completedCount = todos.filter((el) => el.isCompleted === true).length;
+   const activeTask = todos.length - completedCount;
```

Comment:
```suggestion
   const activeTasks = todos.length - completedCount;
```

Пришлось задуматься что это у тебя за активная тачка)